components/vf-analytics-google/vf-analytics-google.js

```diff
@@ -116,21 +122,30 @@ function vfGetMeta(metaName) {
  * Hooks into common analytics tracking
  * @param {object} [vfGaTrackOptions]
  * @param {binary} [vfGaTrackOptions.vfGaTrackPageLoad=true] If true, the function will track the initial page view. Set this to false if you track the page view in your HTML.
+ * @param {string} [vfGaTrackOptions.vfGa4MeasurementId] The GA4 site measurement ID.
  */
 function vfGaInit(vfGaTrackOptions) {
+  // console.log('initing')
   /* eslint-disable no-redeclare*/
   var vfGaTrackOptions = vfGaTrackOptions || {};
   /* eslint-enable no-redeclare*/
   if (vfGaTrackOptions.vfGaTrackPageLoad == null) vfGaTrackOptions.vfGaTrackPageLoad = true;
-
-  // Need help
+  // Need help?
   // How to add dimension to your property
   // https://developers.google.com/analytics/devguides/collection/analyticsjs/custom-dims-mets
   // https://support.google.com/analytics/answer/2709829?hl=en
 
+  if (typeof gtag === "undefined") {
+    // if the site is still using legacy GA, set a dummy gtag function so we don't have to add a bunch of if statements
+    var gtag = function() {};
```

Comment:
I encountered an issue when testing this. When using `var` the `gtag` is hoisted, hence it'll be `undefined` before the check (see https://stackoverflow.com/a/36911203).
A solution is to use the following `window.gtag = function() {}`

Reply:
Hi Ossama! By pure coincidence I've been working through this code this morning, I'll be revising this PR with some fixes that include this issue.